app/screens/UserProfile.js

```diff
@@ -122,27 +142,43 @@ class UserProfile extends Component {
                                         />
                                     </ConfirmationModal>
                                     <Text style={s.username}>{user.username}</Text>
+                                    {admin_rank_int &&
+                                        <View style={s.rankView}>
+                                            <Text style={s.rankText}>{admin_title}</Text>
+                                            <Image source={admin_icon} style={s.rankIcon} />
+                                        </View>
+                                    }
+                                    {!admin_rank_int && contributor_rank_int &&
+                                        <View style={s.rankView}>
+                                            <Text style={s.rankText}>{contributor_rank}</Text>
+                                            <Image source={contributor_icon} style={s.rankIcon} />
+                                        </View>
+                                    }
                                     <Text style={s.member}>{`Joined: ${moment(created_at).format('MMM DD, YYYY')}`}</Text>
                                     <View style={{width:'100%',alignItems:'center'}}>
+                                        <View style={s.statItem}>
+                                            <Text style={s.stat}>Total contributions:</Text>
+                                            <Text style={s.statNum}>{num_total_submissions ? ` ${formatNumWithCommas(num_total_submissions)} ` : ` 0 `}</Text>
+                                        </View>
                                         <View style={s.statItem}>
                                             <Text style={s.stat}>Machines added:</Text>
-                                            <Text style={s.statNum}>{` ${formatNumWithCommas(num_machines_added)} `}</Text>
+                                            <Text style={s.statNum}>{num_machines_added ? ` ${formatNumWithCommas(num_machines_added)} ` : ` 0 `}</Text>
```

Comment:
This totally gets the job done, but it might be nice to abstract it to a function. 

Something like:
```const getStatNum = (stat) => stat ? `${formatNumWithCommas(num_machines_added)} ` : ` 0 ` ```

Then we could just have `getStatNum(num_machines_added)` here.